ZEN-4357 Evaluate Guava dependencies, consider removing where practical

tfesenko opened the pull request:
Guava is a recurring problem with Eclipse. We should consider removing our use of Guava, instead using FP and other features of Java 8.

I removed usages of Guava in KZOE, but I still need to keep a dependency on Guava in our manifest declaration because [JSON Schema Validator uses Guava](https://github.com/java-json-tools/json-schema-core/blob/master/src/main/java/com/github/fge/jsonschema/core/load/configuration/URIDownloadersRegistry.java)

tfesenko committed:
ZEN-4357 Remove dependencies on Guava

tfesenko committed:
Merge branch 'master' into task/ZEN-4357

tfesenko committed:
ZEN-4357 - eliminate usages of Guava in com.reprezen.swagedit.core

tfesenko committed:
Merge branch 'master' into task/ZEN-4357

andylowry reviewed (approved):
Whew! Looks good.

ghillairet commented on com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/assist/JsonContentAssistProcessor.java:
```diff
@@ -196,7 +195,7 @@ protected void clearStatus() {
         final List<ICompletionProposal> result = new ArrayList<>();
 
-        prefix = Strings.emptyToNull(prefix);
+        prefix = "".equals(prefix) ? null : prefix;
```
We should be using `StringUtils.emptyToNull` here.

ghillairet replied:
@tfesenko I did the change.

ghillairet committed:
ZEN-4357 Evaluate Guava dependencies, consider removing where practical

Replace usage of Arrays.asList() with ArrayList constructor where lists are updated, because Arrays.asList() creates an immutable list.
Add missing StringUtils.emptyToNull.

ghillairet committed:
ZEN-4357 Evaluate Guava dependencies, consider removing where practical

Remove references to guava dependency in tests MANIFEST and target platform. Guava is provided by xtend in the target and tests, so it should not be needed to have it declare explicitly.

ghillairet reviewed (approved):
@tfesenko Looks good. I made 2 commits to solve an issue with Arrays.asList(). This method creates an immutable list, and we were using it to create lists that are then updated. 

It also removes guava from tests MANIFEST and target platform as it should not be needed as Guava comes with Xtend.

tfesenko commented on com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/editor/outline/OutlineContentProvider.java:
```diff
@@ -48,7 +48,7 @@ public void inputChanged(Viewer viewer, Object oldInput, Object newInput) {
         }
 
-        List<Object> roots = Arrays.asList();
+        List<Object> roots = new ArrayList<>();
```
Thanks!
